### PR TITLE
Winner list error

### PR DESCRIPTION
**Pull request by robert-waite**
Currently the winners list assumes the player is black... but the player can be randomly white or black.

### CODE REVIEW

File: AlphaGo/training/reinforcement_policy_trainer.py [hunk 1]
```diff
@@ -65,7 +65,12 @@ def do_move(states, moves, X_list, y_list, player_color):
 		done = [st.is_end_of_game for st in states]
 		if all(done):
 			break
-	winners = [st.get_winner() for st in states]
+	winners = list()
```
Comment (wrongu) [inline AlphaGo/training/reinforcement_policy_trainer.py]: Style preference is `[]` instead of `list()`

**Comment by wrongu** [comment]
Great catch. The logic is tricky to follow here, though. You're essentially putting things into the perspective of "learning-player is BLACK" regardless of what their color was during the game. Well, you're making the smallest change that fixes the bug, so that's on us for having it confusing in the first place!

I propose a bigger change throughout the file - replace the `winners` list of player colors with a `won_game` list of booleans that is `True` iff the learning-player won.

`won_game = [st.get_winner() == player_color for st in states]`

This should make `train_batch` cleaner as well; `if won_game` rather than `if winner == -1`

**Comment by robert-waite** [comment]
Adjusted. I think the whole method should change to take into account playing both sides of the board for a particular player in question and the player_color var might not be need to be passed into do_move(). It could maybe be written with the perspective of current player and just requires the caller to manage the sequencing.

File: AlphaGo/training/reinforcement_policy_trainer.py [hunk 2]
```diff
@@ -195,10 +190,10 @@ def save_metadata():
 		if args.verbose:
 			print "Batch {}\tsampled opponent is {}".format(i_iter, opp_weights)
 		# Make training pairs and do RL
-		X_list, y_list, winners = make_training_pairs(player, opponent, features, args.game_batch, board_size)
-		win_ratio = np.sum(np.array(winners) == 1) / float(args.game_batch)
+		X_list, y_list, won_game_list = make_training_pairs(player, opponent, features, args.game_batch, board_size)
+		win_ratio = np.sum(np.array(won_game_list) == True) / float(args.game_batch)
```
Comment (wrongu) [inline AlphaGo/training/reinforcement_policy_trainer.py]: `np.sum(won_game_list)` or `sum(won_game_list)` should also work

**Pull request merged**